File: gherkin/dotnet/Gherkin/Ast/TableCell.cs
namespace Gherkin.Ast
{
    public class TableCell : IHasLocation
    {
        public Location Location { get; private set; }
        public string Value { get; private set; }

        public TableCell(Location location, string value)
        {
            Location = location;
            Value = value;
        }
    }
}

File: gherkin/dart/lib/src/ast/TableCell.dart
import 'package:gherkin/core.dart';
import 'package:gherkin/extensions.dart';
import 'package:gherkin/language.dart';

import 'IHasLocation.dart';

class TableCell implements IHasLocation, INullSafetyObject
{
  static const empty = _InvalidTableCell();

  final Location  _location;

  final String value;

  @override
  Location get location => _location;

  const TableCell(this._location, this.value);

  @override
  bool get isEmpty => false;//_location.isEmpty && value.isEmpty;

  @override
  bool get isNotEmpty => !isEmpty;
}

/// Convenience implementation of an invalid [TableCell] instance.
class _InvalidTableCell extends TableCell
{
  const _InvalidTableCell() : super(Location.empty, Strings.empty) ;

  @override
  bool get isEmpty => true;
}

